rtmp-client/src/main/cpp/librtmp-jni.c

```diff
@@ -78,13 +75,13 @@ Java_net_butterflytv_rtmp_1client_RtmpClient_nativeRead(JNIEnv* env, jobject thi
 
     RTMP *rtmp = (RTMP *) rtmpPointer;
     if (rtmp == NULL) {
-        throwIOException(env, "First call open function");
-        return 0;
+        throwIllegalStateException(env, "RTMP open function has to be called before read");
+        return RTMP_ERROR_IGNORED;
     }
+
     int connected = RTMP_IsConnected(rtmp);
     if (!connected) {
-        throwIOException(env, "Connection to server is lost");
-        return 0;
+        return RTMP_ERROR_CONNECTION_LOST;
```

Comment:
Below on L89, I don't understand how errors that occur during `RTMP_Read` get propagated. Please clarify (sorry for commenting on this line - GitHub doesn't let you comment on a line that hasn't changed)

Reply:
`RTMP_Read` returns the number of bytes read. In case one of the functions it calls, like `Read_1_Packet`, returns an error, it is returned directly - there is a condition to check the return value and return immediately if an error occured.

Earlier, the `r->m_read.status` value, set in `Read_1_Packet` would be used to classify the error in `RTMP_Read`. I have removed that since we do not need the classification anymore. The error is directly returned. I have retained setting the status as it is used to set socket error.